services/placement/placement.go

```diff
@@ -316,3 +354,79 @@ func (s ByIDAscending) Less(i, j int) bool {
 func (s ByIDAscending) Swap(i, j int) {
 	s[i], s[j] = s[j], s[i]
 }
+
+// RemoveInstance removes a given instance from a placement
+func RemoveInstance(list []services.PlacementInstance, instanceID string) []services.PlacementInstance {
+	for i, instance := range list {
+		if instance.ID() == instanceID {
+			return append(list[:i], list[i+1:]...)
+		}
+	}
+	return list
+}
+
+// MarkShardAvailable marks the state of a shard to available
+func MarkShardAvailable(p services.ServicePlacement, instanceID string, shardID uint32) (services.ServicePlacement, error) {
```

Comment:
nit: you could create a markShardAvailable which does everything MarkShardAvailable does now, and MarkShardAvailable just do a clone and then call markShardAvailable, so the markShardAvailable can be reused in MarkAllShardsAsAvailable as well